### ISSUE TITLE
Creating VPC offering: unable to choose compute or network offering

### ISSUE THREAD

**Issue by top-secrett** [apache/cloudstack#7042]
##### ISSUE TYPE

 * Bug Report

##### CLOUDSTACK VERSION

~~~
4.17.1 4.17.2
~~~

##### OS / ENVIRONMENT

OracleLinux 8

##### SUMMARY

I want to create VPC Oferring linked with new network and system offering. The reason is that I need VPC network with specific storage and host tags.
But on VPC creation page I don't see any offering suggestions despite I created system offering and linked network offering with VPC toggle.

##### STEPS TO REPRODUCE

~~~
1. Open Network offering
2. Create new network offering with VPC toggle
3. Open VPC offering
4. Try to add offering with  earlier created  network offering
~~~

##### EXPECTED RESULTS

~~~
Possibility choose compute (network) offering
~~~

##### ACTUAL RESULTS

~~~
There is no any compute (network) offering suggestion.
~~~
![image](https://user-images.githubusercontent.com/92435079/210360133-b5d88e06-f96c-482e-82cf-da838caab38b.png)

**Comment by weizhouapache** [comment]
@top-secrett 
you need to create a VPC offering, not network offering

**Comment by top-secrett** [comment]
@weizhouapache 1. It doesn't matter, there is no compute offering suggestion when I create  VPC offering.
2. And then why CS has VPC toggle in network offering?

**Comment by weizhouapache** [comment]
> @weizhouapache 1. It doesn't matter, there is no compute offering suggestion when I create VPC offering. 2. And then why CS has VPC toggle in network offering?

@top-secrett 
 
to be clarify
(1) vpc offering: offering for vpc
(2) network offering for vpc: offering for vpc tiers
(3) network offering for non-vpc: offering for isolated/shared/L2 networks (depends on network type).

**Comment by weizhouapache** [comment]
@top-secrett 
I confirmed this is a valid bug with UI.
I will fix it.

**Comment by top-secrett** [comment]
@weizhouapache yeah. I've added just a redundant VPC without any tiers and two virtual routers started with defualt compute offering. But vitrual routers started on different clusters. It is a problem for us. And I want to force creation on a certain cluster with storage and hosts tags.
![image](https://user-images.githubusercontent.com/92435079/210413927-6ec50702-4fda-49c3-9600-9d51a2004ed2.png)
![image](https://user-images.githubusercontent.com/92435079/210413974-82a90611-17b5-4184-9795-d1cb256dcc2e.png)

**Comment by weizhouapache** [comment]
> @weizhouapache yeah. I've added just a redundant VPC without any tiers and two virtual routers started with defualt compute offering. But vitrual routers started on different clusters. It is a problem for us. And I want to force creation on a certain cluster with storage and hosts tags. ![image](https://user-images.githubusercontent.com/92435079/210413927-6ec50702-4fda-49c3-9600-9d51a2004ed2.png) ![image](https://user-images.githubusercontent.com/92435079/210413974-82a90611-17b5-4184-9795-d1cb256dcc2e.png)

@top-secrett 
cloudstack tries to deploy redundant routers to different clusters/hosts to have better HA. If a cluster/host is disconnected or down, the VR in another destination will take over.

For the compute offering for VPC offering, it is supported by cloudstack api. 
please refer to https://cloudstack.apache.org/api/apidocs-4.17/apis/createVPCOffering.html
as a workaround, you can also manually update the `service_offering_id` in `vpc_offerings` table in `cloud` database.

**Comment by DensonNgumo** [comment]
> @top-secrett I confirmed this is a valid bug with UI. I will fix it.

Was this bug fixed? I have experienced the same on 4.18.0, which I then upgraded to 4.19 and the bug is still there. I am unable to choose any Network Offering.
![cloudstack_network](https://github.com/apache/cloudstack/assets/11226482/7538af3e-b31a-4eda-8865-70c2fe660bf9)

**Comment by weizhouapache** [comment]
> > @top-secrett I confirmed this is a valid bug with UI. I will fix it.
> 
> Was this bug fixed? I have experienced the same on 4.18.0, which I then upgraded to 4.19 and the bug is still there. I am unable to choose any Network Offering.
> ![cloudstack_network](https://github.com/apache/cloudstack/assets/11226482/7538af3e-b31a-4eda-8865-70c2fe660bf9)
> 

@DensonNgumo 
Your issue seems unrelated.
Is there network offering for L2 network?

**Comment by DensonNgumo** [comment]
> > > @top-secrett I confirmed this is a valid bug with UI. I will fix it.
> > 
> > 
> > Was this bug fixed? I have experienced the same on 4.18.0, which I then upgraded to 4.19 and the bug is still there. I am unable to choose any Network Offering.
> > ![cloudstack_network](https://private-user-images.githubusercontent.com/11226482/337857597-7538af3e-b31a-4eda-8865-70c2fe660bf9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc5OTg5NDMsIm5iZiI6MTcxNzk5ODY0MywicGF0aCI6Ii8xMTIyNjQ4Mi8zMzc4NTc1OTctNzUzOGFmM2UtYjMxYS00ZWRhLTg4NjUtNzBjMmZlNjYwYmY5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDA1NTA0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzN2Q1MzZlODcyYmVkYWYxZjJlZTIxZjI5NWY0YzVmZTczODgwOWY5YjFhZmFiNmQ4NTdjZjk0ZDRjMjFlMWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pA51n2ISMnN5dSn4wTpHNS83lwJomKbO5MuikLflXyo)
> 
> @DensonNgumo Your issue seems unrelated. Is there network offering for L2 network?

Yes, the default offerings are all there as well as custom offerings that I have created. I will open a separate case for this then.
